mp/src/game/client/neo/ui/neo_hud_ammo.cpp

```diff
@@ -137,169 +123,171 @@ void CNEOHud_Ammo::DrawAmmo() const
-			return;
-		}
+	if (digit_as_number && activeWep->UsesClipsForAmmo1())
+	{ // Draw bullets in magazine in number form
+		surface()->DrawSetTextFont(m_hBulletFont);
+		surface()->DrawSetTextPos(digit_xpos + xpos, digit_ypos + ypos);
+		wchar_t bullets[22];
+		V_swprintf_safe(bullets, L"%i/%i", magSizeCurrent, magSizeMax);
+		surface()->DrawPrintText(bullets, (int)(magSizeCurrent == 0 ? 1 : log10(magSizeCurrent) + 1) + (int)(log10(magSizeMax) + 1) + 1);
+		return;
+	}
 
-		auto maxSpaceAvaliableForBullets = digit_max_width;
-		auto bulletWidth = surface()->GetCharacterWidth(m_hBulletFont, *ammoChar);
-		auto plusWidth = surface()->GetCharacterWidth(m_hBulletFont, '+');
-		auto maxBulletsWeCanDisplay = (maxSpaceAvaliableForBullets / bulletWidth);
-		auto maxBulletsWeCanDisplayWithPlus = ((maxSpaceAvaliableForBullets - plusWidth) / bulletWidth);
-		auto bulletsOverflowing = maxBulletsWeCanDisplay < magSizeMax;
+	if (ammoChar == nullptr)
+		return;
 
-		if(bulletsOverflowing)
-		{
-			magSizeMax = maxBulletsWeCanDisplayWithPlus + 1;
-		}
+	const int maxSpaceAvailableForBullets = digit_max_width;
+	const int bulletWidth = surface()->GetCharacterWidth(m_hBulletFont, *ammoChar);
+	const int plusWidth = surface()->GetCharacterWidth(m_hBulletFont, '+');
+	const int maxBulletsWeCanDisplay = bulletWidth == 0 ? 0 : (maxSpaceAvailableForBullets / bulletWidth);
 
-		magSizeMax = min(magSizeMax, 64);
-		char bullets[64]{ '\0' };
-		for(int i = 0; i < magSizeMax; i++)
-		{
-			bullets[i] = *ammoChar;
-		}
+	if (maxBulletsWeCanDisplay == 0)
+		return;
+
+	const int maxBulletsWeCanDisplayWithPlus = bulletWidth == 0 ? 0 : ((maxSpaceAvailableForBullets - plusWidth) / bulletWidth);
+	const bool bulletsOverflowing = maxBulletsWeCanDisplay < magSizeMax;
+
+	if(bulletsOverflowing)
+	{
+		magSizeMax = maxBulletsWeCanDisplayWithPlus + 1;
+	}
 
-		int magAmountToDrawFilled = magSizeCurrent;
+	char bullets[101]; // PZ Mag size plus null character
+	magSizeMax = min(magSizeMax, sizeof(bullets));
+	int i;
+	for(i = 0; i < magSizeMax; i++)
+	{
+		bullets[i] = *ammoChar;
+	}
+	bullets[i] = '\0';
+
+	int magAmountToDrawFilled = magSizeCurrent;
 		
-		if(bulletsOverflowing)
+	if(bulletsOverflowing)
+	{
+		bullets[magSizeMax - 1] = '+';
+
+		if(maxClip == magSizeCurrent)
+		{
+			magAmountToDrawFilled = magSizeMax;
+		} else if(magSizeMax - 1 < magSizeCurrent)
+		{
+			magAmountToDrawFilled = magSizeMax - 1;
+		} else
 		{
-			bullets[magSizeMax - 1] = '+';
-
-			if(maxClip == magSizeCurrent)
-			{
-				magAmountToDrawFilled = magSizeMax;
-			} else if(magSizeMax - 1 < magSizeCurrent)
-			{
-				magAmountToDrawFilled = magSizeMax - 1;
-			} else
-			{
-				magAmountToDrawFilled = magSizeCurrent;
-			}
+			magAmountToDrawFilled = magSizeCurrent;
 		}
+	}
 		
-		wchar_t unicodeBullets[64];
-		g_pVGuiLocalize->ConvertANSIToUnicode(bullets, unicodeBullets, sizeof(unicodeBullets));
+	wchar_t unicodeBullets[100];
```

Comment:
Need allocation of 101 to fit the full `char bullets[101];` contents. Perhaps promote the magic number into a `constexpr auto maxBullets = 100` or similar for reuse?